cmd/admin-rpc-client.go

```diff
@@ -489,3 +567,65 @@ func getValidServerConfig(serverConfigs []serverConfigV13, errs []error) (server
 
 	return configJSON, nil
 }
+
+// Write config contents into a temporary file on all nodes.
+func writeTmpConfigPeers(peers adminPeers, tmpFileName string, config io.Reader) (int64, error) {
+	configBytes, err := ioutil.ReadAll(config)
+	if err != nil {
+		errorIf(err, "Failed to read config from request body.")
+		return 0, err
+	}
+
+	if !globalIsDistXL {
+		return peers[0].cmdRunner.WriteTmpConfig(tmpFileName, configBytes)
+	}
+
+	errs := make([]error, len(peers))
+
+	// Write config into temporary file on all nodes.
+	wg := sync.WaitGroup{}
+	for i, peer := range peers {
+		wg.Add(1)
+		go func(idx int, peer adminPeer) {
+			defer wg.Done()
+			_, errs[idx] = peer.cmdRunner.WriteTmpConfig(tmpFileName, configBytes)
+		}(i, peer)
+	}
+	wg.Wait()
+
+	// Check if the write succeeded in quorum or more number of nodes.
+	rErr := reduceWriteQuorumErrs(errs, nil, len(peers)/2+1)
```

Comment:
We reduce the errors to know how many nodes this operation succeeded on, to be able to proceed with setting of config across all the nodes in a minio distributed setup. If we don't have a majority of success, we fail the operation. Failing on a single failure, will make it difficult to change config when a node is permanently down.